src/pkgcheck/addons/git.py

```diff
@@ -79,14 +76,37 @@ def __init__(self, *args, commit):
         self.commit = commit
 
 
+class GitConfig:
+    """Manages temporary file which holds git config for disabling
+    safe directory feature of git."""
+
+    def __init__(self):
+        fd, self.path = tempfile.mkstemp()
+        print('AAA', self.path)
+        os.write(fd, b'[safe]\n\tdirectory = *\n')
+        os.close(fd)
+
+    @property
+    def config_env(self):
+        # ignore global user and system git config, but disable safe.directory
+        return ImmutableDict({
+            'GIT_CONFIG_GLOBAL': self.path,
+            'GIT_CONFIG_SYSTEM': '',
+        })
+
+    def close(self):
+        os.unlink(self.path)
+
+
 class GitLog:
     """Iterator for decoded `git log` line output."""
 
     def __init__(self, cmd, path):
         self._running = False
+        self.safe_directory = GitConfig()
```

Comment:
[nit] I would maybe rename this variables to `self.git_config` so it'd be
```python
env=self.git_config.config_env
```
in the call to `subprocens.Popen`

Reply:
done